### ISSUE TITLE
Connect statement errors with "Character set '255' unsupported"

### ISSUE THREAD

**Issue by jchambers-singlestore** [singlestore-labs/singlestore-python#1]
Python file
```
(dbt-env) ubuntu@h1:~$ cat connect.py
import singlestore as s2

HOST = 'h2.singlestore.unx.com'
PORT = 3306
USER = 'root'
PASSWORD = 'Pass@word1'
DATABASE = 'db'

conn = s2.connect(host=HOST, port=PORT, user=USER, password=PASSWORD, database=DATABASE, charset='utf8')
```

Error
```
(dbt-env) ubuntu@h1:~$ python3 connect.py
Traceback (most recent call last):
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/singlestore/connection.py", line 916, in __init__
    self._conn = self._driver.connect()
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/singlestore/drivers/base.py", line 63, in connect
    conn = self.dbapi.connect(**params)
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/pooling.py", line 286, in connect
    return CMySQLConnection(*args, **kwargs)
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/connection_cext.py", line 101, in __init__
    self.connect(**kwargs)
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/abstracts.py", line 1099, in connect
    self._post_connection()
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/abstracts.py", line 1071, in _post_connection
    self.set_charset_collation(self._charset_id)
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/abstracts.py", line 1016, in set_charset_collation
    ) = CharacterSet.get_charset_info(charset)
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/constants.py", line 775, in get_charset_info
    info = cls.get_default_collation(charset)
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/mysql/connector/constants.py", line 746, in get_default_collation
    raise ProgrammingError(f"Character set '{charset}' unsupported")
mysql.connector.errors.ProgrammingError: Character set '255' unsupported

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/connect.py", line 14, in <module>
    conn = s2.connect('root:Pass@word1@h2.singlestore.unx.com:3306/db')
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/singlestore/connection.py", line 1223, in connect
    return Connection(**dict(locals()))
  File "/home/ubuntu/dbt-env/lib/python3.10/site-packages/singlestore/connection.py", line 918, in __init__
    raise self._driver.convert_exception(exc)
singlestore.exceptions.ProgrammingError: -1: Character set '255' unsupported
```

Version info
```
(dbt-env) ubuntu@h1:~$ pip list | grep singlestore
dbt-singlestore          1.1.2
singlestore              0.1.0
```
```
(dbt-env) ubuntu@h1:~$ python3 --version
Python 3.10.4
```
```
singlestore> select @@memsql_version;
+------------------+
| @@memsql_version |
+------------------+
| 7.8.8            |
+------------------+
```

Also tested these python connect statements which resulted in the same error as above.
```
#conn = s2.connect('h2.singlestore.unx.com/db?local_infile=True&charset=utf8mb4&password=Pass@word1&user=root')
#conn = s2.connect(host=HOST, port=PORT, user=USER, password=PASSWORD, database=DATABASE, charset='utf8mb4')
#conn = s2.connect('root:Pass@word1@h2.singlestore.unx.com:3306/db')
```

**Comment by jchambers-singlestore** [comment]
> Can you include mysql-connector in your `pip list` as well? That's the backend driver that is doing the actual communication with the server.

(dbt-env) ubuntu@h1:~$ pip list | grep -E 'my|sing'
dbt-singlestore          1.1.2
mypy-extensions          0.4.3
mysql-connector-python   8.0.30
pyparsing                3.0.9
singlestore              0.1.0

**Comment by kesmit13** [comment]
Have you tried connecting directly using mysql-connector-python or some other MySQL client? The SingleStore package simply passes those arguments on. I haven't been able to reproduce this issue, so I'm just trying to get more information.

**Comment by kesmit13** [comment]
This appears to be an issue with the latest version of mysql.connector. We are actually in the process of changing the driver in the SingleStore Python package to PyMySQL, which I believe will solve this issue. We are trying to get the new release out in a couple of days.

**Comment by kesmit13** [comment]
We've verified that this is a bug in the mysql.connector package. The current version of the SingleStore Python package (v0.2.0) no longer uses that driver, so if you change to that version, it should work. Note that the package name changed to singlestoredb rather than singlestore. If you want to work around the issue, you can do the following
```
conn = s2.connect(host="...", port=..., user="...", password="...")
conn.set_charset_collation(charset="utf8")
```

**Comment by jchambers-singlestore** [comment]
Fixed with version 2.0, using `singlestoredb` package instead of `singlestore`.

Thanks for the quick turnaround!
